account_journal_general_sequence/tests/test_numbering.py

```diff
@@ -85,3 +85,20 @@ def test_install_no_entry_number(self):
         invoice.action_post()
         # Ensure there's no entry number
         self.assertFalse(invoice.entry_number)
+
+    @users("admin")
+    def test_new_company_journal(self):
+        cmp = self.env["res.company"].create({"name": "Company100", "currency_id": 2})
+        self.env["account.journal"].create(
+            {
+                "name": "Journal_test100",
+                "code": "code100",
+                "type": "bank",
+                "company_id": cmp.id,
+            }
+        )
+        self.env.company_id = cmp.id
```

Comment:
This pattern can introduce a problem. We are not supposed to alter the company like that. The company is supposed to be obtained from the context, and the context is supposed to be frozen. If you do it like this, it can have unwanted side effects.

Instead, create a new env and assign it:

```suggestion
        self.env = self.env(context=dict(self.env.context, allowed_company_ids=cmp.ids))
```